code/game/objects/structures/windows/window.dm

```diff
@@ -90,6 +92,24 @@
 	if (flags_1 & ON_BORDER_1)
 		AddElement(/datum/element/connect_loc, loc_connections)
 
+/obj/structure/window/mouse_drop_receive(atom/dropping, mob/user, params)
+	. = ..()
+	if (added_leaning)
```

Comment:
```suggestion
	if (GetComponent(/datum/component/leanable))
```

And do the same for the other one and remove the var

Reply:
GetComponent is generally considered ugly and should be avoided when possible

Reply:
According to who? It's preferred to storing components as refs and is best for boolean checking in this case.

Just do a search text in VS code on how many places this proc is used before making such statements